### ISSUE TITLE
v0.14.1: Impossible to send a command remotely via SSH since $TERM variable must be set

### ISSUE THREAD

**Issue by elsagermann** [spack/spack#15755]
Hello,
I was trying to install a package using the command `spack install <spec>` **via Jenkins** using the new spack release v0.14.1 and ran into the following error:

### Spack version

```console
$ spack --version
0.14.1
```

### Steps to reproduce the issue

```console
$ ssh user@daint.cscs.ch 'spack install <spec>'
...
```

### Error Message

```

Traceback (most recent call last):
  File "/project/g110/spack/user/admin-daint/spack/bin/spack", line 64, in <module>
    sys.exit(spack.main.main())
  File "/project/g110/spack/user/admin-daint/spack/lib/spack/spack/main.py", line 770, in main
    if spack.config.get('config:debug'):
  File "/project/g110/spack/user/admin-daint/spack/lib/spack/spack/config.py", line 650, in get
    return config.get(path, default, scope)
  File "/project/g110/spack/user/admin-daint/spack/lib/spack/llnl/util/lang.py", line 552, in __getattr__
    return getattr(self.instance, name)
  File "/project/g110/spack/user/admin-daint/spack/lib/spack/llnl/util/lang.py", line 548, in instance
    self._instance = self.factory()
  File "/project/g110/spack/user/admin-daint/spack/lib/spack/spack/config.py", line 632, in _config
    _add_platform_scope(cfg, ConfigScope, name, path)
  File "/project/g110/spack/user/admin-daint/spack/lib/spack/spack/config.py", line 585, in _add_platform_scope
    platform = spack.architecture.platform().name
  File "/project/g110/spack/user/admin-daint/spack/lib/spack/llnl/util/lang.py", line 178, in _memoized_function
    func.cache[args] = func(*args)
  File "/project/g110/spack/user/admin-daint/spack/lib/spack/spack/architecture.py", line 516, in platform
    return platform_cls()
  File "/project/g110/spack/user/admin-daint/spack/lib/spack/spack/platforms/cray.py", line 63, in __init__
    _target = self._default_target_from_env()
  File "/project/g110/spack/user/admin-daint/spack/lib/spack/spack/platforms/cray.py", line 121, in _default_target_from_env
    env={'TERM': os.environ['TERM']},
  File "/usr/lib64/python3.6/os.py", line 669, in __getitem__
    raise KeyError(key) from None
KeyError: 'TERM'

```

### Information on your system

Static hostname: daint102elf.spec:
Icon name: computer-server
Chassis: server
Machine ID: 3c702e3731a14a76aecdd784a323529epy" 
Boot ID: 64c4c855a3a940e7a1846942dd9a2109
Operating System: SUSE Linux Enterprise Server 15
CPE OS Name: cpe:/o:suse:sles:15
Kernel: Linux 4.12.14-150.35-default
Architecture: x86-64

### General information

- [x] I have run `spack --version` and reported the version of Spack
- [x] I have searched the issues of this repo and believe this is not a duplicate

I found that the problem came from the fact that Jenkins is sending its commands remotely via SSH and not opening a terminal, thus not setting the $TERM variable. Setting the variable to a default value in the Jenkins script solves the problem, but it would be great not to have to do that manually.
The problem is only happening with the new released version (v0.14.1), until then I never had this problem. Also, I found that some servers do still find a way to set the $TERM variable, but on Piz Daint for example it is not working.

Thanks a lot in advance for any answer

**Comment by elsagermann** [comment]
This issue is closely related to this issue [15620](https://github.com/spack/spack/issues/15620), but might be still useful so I will let it here for the moment.

**Comment by adamjstewart** [comment]
This issue was fixed by #15630, and a related issue will be fixed by #15723. I'll close this issue when v0.14.2 is released (hopefully soon).

**Comment by adamjstewart** [comment]
This has been fixed now that 0.14.2 has been released.
